### PR TITLE
[X86][MC] Support Enc/Dec for EGPR for promoted BMI instructions

### PR DESCRIPTION
**Pull request by XinWang10**
R16-R31 was added into GPRs in https://github.com/llvm/llvm-project/pull/70958,
This patch supports the encoding/decoding for promoted BMI instructions in EVEX space.

RFC: https://discourse.llvm.org/t/rfc-design-for-apx-feature-egpr-and-ndd-support/73031/4

### CODE REVIEW

File: llvm/lib/Target/X86/X86InstrMisc.td [hunk 1]
```diff
@@ -1292,56 +1301,50 @@ let Predicates = [HasBMI] in {
...
-                    Sched.ReadAfterFold]>;
+multiclass bmi4VOp3_base<bits<8> opc, string mnemonic, RegisterClass RC,
+                         X86MemOperand x86memop, SDPatternOperator OpNode,
+                         PatFrag ld_frag, X86FoldableSchedWrite Sched,
+                         string Suffix = ""> {
+  def rr#Suffix : I<opc, MRMSrcReg4VOp3, (outs RC:$dst), (ins RC:$src1, RC:$src2),
+                    !strconcat(mnemonic, "\t{$src2, $src1, $dst|$dst, $src1, $src2}"),
+                    [(set RC:$dst, (OpNode RC:$src1, RC:$src2)), (implicit EFLAGS)]>,
+                  T8PS, VEX, Sched<[Sched]>;
+let mayLoad = 1 in
```
Comment (KanRobert) [inline llvm/lib/Target/X86/X86InstrMisc.td:1312]: Why add `mayLoad = 1` here?

**Review by KanRobert — APPROVED** [review]
LGTM with one question

File: llvm/lib/Target/X86/X86InstrMisc.td (hunk 1 above)
Comment (XinWang10) [reply llvm/lib/Target/X86/X86InstrMisc.td:1312]: When I tried to generate mca test before, it miss the this attribute mayload if we don't explicitliy set it.

**Comment by KanRobert** [comment]
Need rebase

**Pull request merged**